pkgs/by-name/vc/vcpkg-tool/package.nix

```diff
@@ -47,31 +51,139 @@ stdenv.mkDerivation (finalAttrs: {
     "-DVCPKG_DEPENDENCY_CMAKERC=ON"
   ];
 
-  postFixup = let
+
+  passAsFile = [ "vcpkgWrapper" ];
+  vcpkgWrapper = let
     # These are the most common binaries used by vcpkg
-    # Extra binaries can be added via overlay when needed
+    # Extra binaries can be added through override when needed
     runtimeDeps = [
       cacert
       cmake
+      curl
       git
       gzip
       meson
       ninja
       openssh
       python3
+      unzip
       zip
       zstd
     ] ++ extraRuntimeDeps;
+
+    # Apart from adding the runtime dependencies to $PATH,
+    # the wrapper will also override these arguments by default.
+    # This is to ensure that the executable does not try to
+    # write to the nix store. If the user tries to set any of the
+    # arguments themself, the wrapper will detect that the
+    # arguments are present, and prefer the user-provided value.
+    #
+    # It is also possible to override the cli arguments by
+    # settings either of the nix-specific environment variables.
+    argsWithDefault = [
+      {
+        arg = "--downloads-root";
+        env = "NIX_VCPKG_DOWNLOADS_ROOT";
+        default = "$NIX_VCPKG_WRITABLE_PATH/downloads";
+      }
+      {
+        arg = "--x-buildtrees-root";
+        env = "NIX_VCPKG_BUILDTREES_ROOT";
+        default = "$NIX_VCPKG_WRITABLE_PATH/buildtrees";
+      }
+      {
+        arg = "--x-packages-root";
+        env = "NIX_VCPKG_PACKAGES_ROOT";
+        default = "$NIX_VCPKG_WRITABLE_PATH/packages";
+      }
+      {
+        arg = "--x-install-root";
+        env = "NIX_VCPKG_INSTALL_ROOT";
+        default = "$NIX_VCPKG_WRITABLE_PATH/installed";
+      }
+    ];
   in ''
-    wrapProgram $out/bin/vcpkg --prefix PATH ${lib.makeBinPath runtimeDeps}
+    #!${runtimeShell}
+
+    get_vcpkg_path() {
+      if [[ -n $NIX_VCPKG_WRITABLE_PATH ]]; then
+          echo "$NIX_VCPKG_WRITABLE_PATH"
+      elif [[ -n $XDG_CACHE_HOME ]]; then
+          echo "$XDG_CACHE_HOME/vcpkg"
+      elif [[ -n $HOME ]]; then
+          echo "$HOME/.vcpkg/root"
+      elif [[ -n $TMP ]]; then
+          echo "$TMP"
+      else
+          echo "/tmp"
+      fi
+    }
+
+    NIX_VCPKG_WRITABLE_PATH=$(get_vcpkg_path)
+
+    ${lib.concatMapStringsSep "\n" ({ env, default, ... }: ''${env}=''${${env}-"${default}"}'') argsWithDefault}
+
+    export PATH="${lib.makeBinPath runtimeDeps}''${PATH:+":$PATH"}"
+
+    ARGS=( "$@" )
```

Comment:
The problem seems to originate from here. I changed the script to output the values. Here's what I did:
```
ARGS=( "$@" )
echo "REALLY_ALL_ARGS $@"
echo "ARGS ${ARGS}"
```
Here's the output I see in my project:
```
REALLY_ALL_ARGS install --triplet wasm32-unknown-emscripten --vcpkg-root /Users/grc/Workspace/github/nixpkgs/result/share/vcpkg --x-wait-for-lock --x-manifest-root=/Users/grc/Workspace/software/pixmob/apps/visualizer/visualizer-web/packages/vtx --x-install-root=/Users/grc/Workspace/software/pixmob/apps/visualizer/visualizer-web/packages/vtx/build/dev-emscripten/vcpkg_installed --host-triplet=aarch64-darwin-clang --overlay-triplets=/Users/grc/Workspace/software/pixmob/apps/visualizer/visualizer-web/packages/vtx/triplets
ARGS install
```
Here `ARGS` should not be just `install`, it should be an array with all the args above, if I understood the scripts correctly.

Reply:
This seems to be an issue indeed. `$ARRAY` only yields the first element of the array, while `${ARRAY[@]}` yields all elements. I seem to have overlooked it for `FINAL_NONMODIFIED_ARGS`, making it so only the first of the non-special arguments are forwarded from the wrapper. I'll push a fix. Does it resolve the overlay issues as well?

Reply:
The updated version seems to work as expected. I approve the changes!